src/common/auth.ts

```diff
@@ -0,0 +1,42 @@
+import { compare, genSalt, hash } from "bcrypt";
+import { psqlClient } from "./database";
+import { sendError, ErrorCodes } from "./error";
+import { User } from "./types";
+import { randomBytes } from "crypto";
+import { salt } from "./constants";
+
+export async function checkCredentials(
+    email: string,
+    password: string
+): Promise<{
+    success: boolean;
+    id?: bigint;
+    error?: "invalidCredentials" | "invalidTOTP" | "accountNotActivated";
+}> {
```

Comment:
why not use a dependent type?
```suggestion
): Promise<{
    success: true;
    id: bigint;
} | {
    success: false;
    error: "invalidCredentials" | "invalidTOTP" | "accountNotActivated";
}> {
```

Reply:
didnt know this was doable